source/client/options_impl.cc

```diff
@@ -496,6 +503,17 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv) {
       throw MalformedArgvException(e.what());
     }
   }
+  if (!request_source_plugin_config.getValue().empty()) {
+    try {
+      request_source_plugin_config_.emplace(envoy::config::core::v3::TypedExtensionConfig());
```

Comment:
This is basically just a copy of the above code, which leads me to ask:
Would it be possible / help readability if we abstracted this recipe into a private helper?  SetFieldFromJSON or something?